snorkeling_helper/label_functions/dependency_cluster/global_network_datafile_generator.py

```diff
@@ -0,0 +1,251 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:snorkeling]
+#     language: python
+#     name: conda-env-snorkeling-py
+# ---
+
+# # Biclustering of Dependency Paths for Biomedical Realtionship Extraction
+
+# A global network of biomedical relationships derived from text
+
+# +
+import networkx as nx
+import pandas as pd
+import spacy
+from tqdm import tqdm
+
+tqdm.pandas(desc="LF progressbar")
+# -
+
+nlp = spacy.load("en_core_web_sm")
+
+
+def convert_dep_path(dataframe_row):
+    is_first_entity_first = int(
+        dataframe_row["first_entity_location"].split(",")[1]
+    ) < int(dataframe_row["second_entity_location"].split(",")[0])
+    seen_first_entity = False
+    build_string = []
+    for token in dataframe_row.sentence.split(" "):
+        if token == dataframe_row["second_entity_name"]:
+            if is_first_entity_first and seen_first_entity:
+                token = "end_entity"
+
+            if not is_first_entity_first:
+                token = "start_entity"
+
+        if token == dataframe_row["first_entity_name"]:
+            if is_first_entity_first:
+                token = "start_entity"
+                seen_first_entity = True
+            else:
+                token = "end_entity"
+
+        build_string.append(token.lower())
+
+    # Skip sentences with multiple entities
+    sen_text = " ".join(build_string).lower()
+    if sen_text.count("start_entity") > 1 or sen_text.count("end_entity") > 1:
+        return ""
+
+    try:
+        doc = nlp(sen_text)
+        dep_graph = nx.Graph()
+        for token in doc:
+            for child in token.children:
+                dep_graph.add_edge(
+                    "{0}".format(token), "{0}".format(child), dep=child.dep_
+                )
+
+        path = nx.shortest_path(dep_graph, "start_entity", "end_entity")
+        pathGraph = nx.path_graph(path)
+
+        return " ".join(
+            [
+                f"{ea[1]}|{dep_graph.edges[ea[0], ea[1]]['dep']}|{ea[0]}"
+                for ea in pathGraph.edges()
+            ]
+        )
+
+    except nx.NetworkXNoPath:
+        return ""
+    except nx.NodeNotFound:
+        return ""
+
+
+# # Chemical-Disease
```

Comment:
If you think it might be helpful, maybe a short description of what you do here in each section.